odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,98 @@
+
+    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920", readonly=False)
+
+    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024", readonly=False)
+    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512", readonly=False)
+    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256", readonly=False)
+    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128", readonly=False)
+
+    def _compute_avatar(self, avatar_field, image_field):
+        for record in self:
+            record[avatar_field] = record[image_field] or record._get_one_avatar(avatar_field, image_field)
+
+    def _get_one_avatar(self, avatar_field, image_field):
+        self.ensure_one()
+        if self[self._rec_name]:
+            return self._generate_avatar_svg(self[self._rec_name])
+        return self._get_avatar_placeholder()
+
+    @api.depends(lambda self: [self._rec_name, 'image_1920'])
+    def _compute_avatar_1920(self):
+        self._compute_avatar('avatar_1920', 'image_1920')
+
+    @api.depends(lambda self: [self._rec_name, 'image_1024'])
+    def _compute_avatar_1024(self):
+        self._compute_avatar('avatar_1024', 'image_1024')
+
+    @api.depends(lambda self: [self._rec_name, 'image_512'])
+    def _compute_avatar_512(self):
+        self._compute_avatar('avatar_512', 'image_512')
+
+    @api.depends(lambda self: [self._rec_name, 'image_256'])
+    def _compute_avatar_256(self):
+        self._compute_avatar('avatar_256', 'image_256')
+
+    @api.depends(lambda self: [self._rec_name, 'image_128'])
+    def _compute_avatar_128(self):
+        self._compute_avatar('avatar_128', 'image_128')
+
+    def _get_avatar_seed(self):
+        return str(self.create_date.timestamp())
+
+    def _generate_avatar_svg(self, name):
+        encoded_name = sha512(self._get_avatar_seed().encode('utf-8'))
+        initials = html_escape(self._generate_avatar_initials(name))
+        bgcolor = html_escape(self._generate_avatar_bgcolor(encoded_name))
+        return b64encode(bytes((
+            "<?xml version='1.0' encoding='UTF-8' ?>"
+            "<svg height='180' width='180' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'>"
+            f"<rect fill='{bgcolor}' height='180' width='180'/>"
+            f"<text fill='#ffffff' font-size='96' text-anchor='middle' x='90' y='125' font-family='sans-serif'>{initials}</text>"
+            "</svg>"
+        ), 'utf-8'))
+
+    def _get_avatar_placeholder(self):
+        image_path = get_resource_path('base', 'static/img', 'avatar_grey.png')
+        return b64encode(file_open(image_path, 'rb').read())
+
+    def _generate_avatar_initials(self, name):
+        return name[0].upper()
+
+    def _generate_avatar_bgcolor(self, encoded_name):
```

Comment:
Could be simplified in a single method (no need to call ma method each time you want to modify a single value)

Reply:
Wouldn't moving this method to `_generate_avatar_svg` be too much pollution since `_generate_avatar_bgcolor` has multiple lines?

Reply:
You don't have to keep this level of complexity... in the end, you have a method per line which makes code quite impossible to understand without jumping from method to method. Keeping it simple, computing all bg color stuff could be done quite easily, especially you don't need to allow overriding every little parameter.

Reply:
The idea of keeping external small methods is that people don't need to care for how i implemented the HSL computation. I feel having it all in one method would add much more noise than keeping it separate.

Reply:
Actually people should not care about the whole method. Either they use the avatar, either they want to have something custom and will get rid of your method completely. Nobody will tweak a single parameter.

I don't think having 5 lines for hsl computation is "noisy". Either you want to read it, either not. Hiding it behind sub methods just make code more difficult to read when you want to read it.

Keep it simple. Don't do submethods just for some side parameters. If in a near future we want to ease its customization, we will do it when having real use cases, not before.